pkg/extensions/sync/utils.go

```diff
@@ -89,6 +100,284 @@ func getFileCredentials(filepath string) (CredentialsFile, error) {
 	return creds, nil
 }
 
+func getHTTPClient(regCfg *RegistryConfig, credentials Credentials, log log.Logger) (*resty.Client, error) {
+	client := resty.New()
+
+	if regCfg.CertDir != "" {
+		log.Debug().Msgf("sync: using certs directory: %s", regCfg.CertDir)
+		clientCert := path.Join(regCfg.CertDir, "client.cert")
+		clientKey := path.Join(regCfg.CertDir, "client.key")
+		caCertPath := path.Join(regCfg.CertDir, "ca.crt")
+
+		caCert, err := ioutil.ReadFile(caCertPath)
+		if err != nil {
+			log.Error().Err(err).Msg("couldn't read CA certificate")
+			return nil, err
+		}
+
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
+
+		client.SetTLSClientConfig(&tls.Config{RootCAs: caCertPool})
+
+		cert, err := tls.LoadX509KeyPair(clientCert, clientKey)
+		if err != nil {
+			log.Error().Err(err).Msg("couldn't read certificates key pairs")
+			return nil, err
+		}
+
+		client.SetCertificates(cert)
+	}
+
+	// nolint: gosec
+	if regCfg.TLSVerify != nil && !*regCfg.TLSVerify {
+		client.SetTLSClientConfig(&tls.Config{InsecureSkipVerify: true})
+	}
+
+	if credentials.Username != "" && credentials.Password != "" {
+		log.Debug().Msgf("sync: using basic auth")
+		client.SetBasicAuth(credentials.Username, credentials.Password)
+	}
+
+	return client, nil
+}
+
+func syncCosignSignature(client *resty.Client, storeController storage.StoreController,
+	regURL url.URL, repo, digest string, log log.Logger) error {
+	log.Info().Msg("syncing cosign signatures")
+
+	getCosignManifestURL := regURL
+
+	cosignEncodedDigest := strings.Replace(digest, ":", "-", 1) + ".sig"
```

Comment:
does cosign have any utility libs/apis to do this for us?

Reply:
they have one but is not exported...